src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2947,8 +2947,6 @@ CopyRelationRestrictionContext(RelationRestrictionContext *oldContext)
 
 		/* not copyable, but readonly */
 		newRestriction->plannerInfo = oldRestriction->plannerInfo;
-		newRestriction->prunedShardIntervalList =
```

Comment:
Do we want to set this to NULL? Also, we can take this to another commit as you noted.